### ISSUE TITLE
RFE: way to disable blocking temporarily (one page load or limited time)

### ISSUE THREAD

**Issue by vassudanagunta** [gorhill/uBlock#3178]
### Describe the issue

There is often a need for a **one-time** or **temporary** disabling of blocking, to get access to content on a page just once. The only way to do that today is to 

1. disable uBlock
2. refresh the page
3. remember to re-enable uBlock after it loads. 

Not only is this three steps, the last one is easy to forget. 

### Steps for anyone to reproduce the issue

1. Go to a page that has video for which uBlock filters prevent the video from playing. For example:  http://www.nytimes.com/video
2. Pressing play has no effect while uBlock is enabled. 
3. Try to find the easiest way to play the video, but without permanently disabling uBlock on the site.

### Your settings

(I'm pretty sure my settings aren't relevant to this RFE)

##### Your filter lists

(I'm pretty sure my filter lists aren't relevant to this RFE, except perhaps to reproduce with the nytimes example I provided.)

**Comment by gorhill** [comment]
> Go to a page that has video for which uBlock filters prevent the video from playing. For example: http://www.nytimes.com/video

Video play fine here without having to disable uBO.

**Comment by vassudanagunta** [comment]
I probably have more filters enabled. I can figure out which ones and add to the issue, but that link was just for illustrative purposes. Even without, I think you can understand the scenario I'm talking about? The need to sometimes disable blocking for just one view of the content?

**Comment by Hrxn** [comment]
Yes, but disabling uBlock (the standard method, from the menu etc.) is already site specific?
Not sure what you mean?

**Comment by vassudanagunta** [comment]
@Hrxn I'm not asking for _per-site_ disabling. I'm asking for temporary disabling. A one-time disabling. 

@gorhill Have you never run into this situation? Where you normally don't care that some feature of a site is broken due to blocking (e.g. the nytimes videos for me), you'd rather keep your strict filters on, and don't have the time or expertise to figure out the filter bug (me, not you of course), and all you want to do is let the page load ONE TIME without blocking enabled.  

Obviously the disable feature exists because the filters break some sites and it is not worth tracing the filter problem. I'm saying it would be useful (to many people I think) to be able to disable temporarily.

**Comment by Hrxn** [comment]
Ah, okay.
Yeah, sounds reasonable to me..

**Comment by jspenguin2017** [comment]
I'd like to point out that temporary white list can be very useful on those CoinHive Captchas, when you just want to get pass that Captcha but you don't want to get the whole domain permanently white listed. 
For example, on this page: `https://cnhv.co/92vq`, if I want the real link behind the Captcha, I would white list `cnhv.co`, let it run, and... it redirects me to the real link and I have to enter settings page to remove the white list.

**Comment by vassudanagunta** [comment]
uBO has the first two. UX-wise it makes sense to add the third:

action on uBO on/off button | enable/disable (whitelist)...
-----| -----
click | current site
`Ctrl`-click | current page
`Alt`-click (or `shift`-click?) |  current page for 15 seconds (or until it fully loads, or whatever is most feasible).

**Comment by vassudanagunta** [comment]
@gorhill I identified the third-party filter to reproduce the specific example. Though note this is not a filtering flaw issue. It's an RFE. I added a recommendation to the issue description to make it clearer for everyone.

**Comment by konradpapala** [comment]
I fully agree, I don't understand why this thing doesn't have hundreds of votes and isn't done a decade ago :) I mean, I see an article about anything, it has some videos and thing, or even this dialog saying "you have an ad blocking software installed.... please.....". 

I just want to see and read this one article and forget it. Forget the article and the whole site. I probably will never get back to it. 

I don't want to 'ctrl+click' click to whitelist current site, because with time my whitelist grows to hundreds of line of websites I just wanted to read once o.O

Just one addition to the @vassudanagunta suggestion: the Alt-click (or Shift-click?) could be user definable (selectable), and then we could have disable: 

"current page once" , or
 
"current page for the full current **session**"   - i.e. until the browser is reset and I would personally choose this option.

**Comment by sanjayen** [comment]
I wish this was added to uBlock Origin at the earliest. I have to disable/ enable every time I do an online payment. Something or the other always breaks on the payment gateway page and my transaction fails. Please do provide this feature.

**Comment by BogudanPro** [comment]
Any news about the possibility of the implementation?
